test/metabase/sync/analyze/fingerprint_test.clj

```diff
@@ -272,23 +274,26 @@
                   :failed-fingerprints    0
                   :updated-fingerprints   1
                   :fingerprints-attempted 1}
-                 (#'sync.fingerprint/fingerprint-table! (t2/select-one :model/Table :id (data/id :venues)) [field])))
+                 (#'sync.fingerprint/fingerprint-fields! (t2/select-one :model/Table :id (data/id :venues)) [field])))
           (is (= {:fingerprint         {:experimental {:fake-fingerprint? true}}
                   :fingerprint_version 3
                   :last_analyzed       nil}
                  (into {} (t2/select-one [:model/Field :fingerprint :fingerprint_version :last_analyzed] :id (u/the-id field))))))))))
 
-(deftest test-fingerprint-failure
-  (testing "if fingerprinting fails, the exception should not propagate"
-    (with-redefs [sync.fingerprint/fingerprint-table! (fn [_ _] (throw (Exception. "expected")))]
-      (is (= (sync.fingerprint/empty-stats-map 0)
-             (sync.fingerprint/fingerprint-fields! (t2/select-one :model/Table :id (data/id :venues))))))))
+(deftest fingerprint-failure-test
+  (testing "if fingerprinting fails, the exception should be caught and included in the stats map"
+    (mt/test-drivers (mt/normal-drivers)
+      (with-redefs [sync.fingerprint/fingerprint-fields! (fn [_ _] (throw (Exception. "expected")))
+                    sync.fingerprint/*refingerprint?* true]
+        (is (=? (assoc (sync.fingerprint/empty-stats-map 0)
+                       :throwable #(instance? Exception %))
```

Comment:
I think you can `let` the Exception above, and test equality here. If not, it would be good to check `(is (= (ex-message %) "expected"))`